llmfoundry/registry.py

```diff
@@ -137,6 +138,61 @@
     description=_dataloaders_description,
 )
 
+_dataset_replication_validators_description = (
+    """Validates the dataset replication args.
+    Args:
+        cfg (DictConfig): The dataloader config.
+        tokenizer (PreTrainedTokenizerBase): The tokenizer
+        device_batch_size (Union[int, float]): The device batch size.
+    Returns:
+        replication_factor (int): The replication factor for dataset.
+        ds_batch_size (int): The dataset device batch size."""
+)
+dataset_replication_validators = create_registry(
+    'llmfoundry',
+    'dataset_replication_validators',
+    generic_type=Callable[
+        [DictConfig, PreTrainedTokenizerBase, Union[int, float]], Tuple[int,
+                                                                        int]],
+    entry_points=True,
+    description=_dataset_replication_validators_description,
+)
+
+_collators_description = (
+    """Returns the data collator.
+    Args:
+        cfg (DictConfig): The dataloader config.
+        tokenizer (PreTrainedTokenizerBase): The tokenizer
+        ds_batch_size (Union[int, float]): The dataset device batch size.
+    Returns:
+        collate_fn  (Any): The collate function.
+        dataloader_batch_size (Optional[int]): The dataloader batch size, used for packing. Only finetuning collator returns this."""
```

Comment:
they should all return this. if it doesn't need to change it, it should just be a no op on the input, right?

Reply:
Right. Fixed.